notebooks/community/aisamples/AIsample - Uplift Modelling.ipynb

```diff
@@ -0,0 +1,882 @@
+    "    title={A Large Scale Benchmark for Uplift Modeling},\n",
+    "    publisher = {ACM},\n",
+    "    booktitle = {Proceedings of the AdKDD and TargetAd Workshop, KDD, London,United Kingdom, August, 20, 2018},\n",
+    "    year = {2018}\n",
+    "    }\n",
+    "    ```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "if not IS_CUSTOMER_DATA:\n",
+    "    # Download demo data files into lakehouse if not exist\n",
+    "    remote_url = \"http://go.criteo.net/criteo-research-uplift-v2.1.csv.gz\"\n",
+    "    download_file = \"criteo-research-uplift-v2.1.csv.gz\"\n",
+    "\n",
+    "    # For this demo, we first check if the dataset files are already prepared in the default lakehouse. If not, we'll download the dataset.\n",
+    "    import os\n",
+    "    import requests\n",
+    "\n",
+    "    if not os.path.exists(\"/lakehouse/default\"):\n",
+    "        # ask user to add a lakehouse if no default lakehouse added to the notebook.\n",
+    "        # a new notebook will not link to any lakehouse by default.\n",
+    "        raise FileNotFoundError(\n",
+    "            \"Default lakehouse not found, please add a lakehouse for the notebook.\"\n",
+    "        )\n",
+    "    else:\n",
+    "        # check if the needed files are already in the lakehouse, try to download if not.\n",
+    "        # raise an error if downloading failed.\n",
+    "        os.makedirs(f\"/lakehouse/default/{DATA_FOLDER}/raw/\", exist_ok=True)\n",
+    "\n",
+    "        if not os.path.exists(f\"/lakehouse/default/{DATA_FOLDER}/raw/{DATA_FILE}\"):\n",
+    "            try:\n",
+    "                r = requests.get(f\"{remote_url}\", timeout=30)\n",
+    "                with open(\n",
+    "                    f\"/lakehouse/default/{DATA_FOLDER}/raw/{download_file}\", \"wb\"\n",
+    "                ) as f:\n",
+    "                    f.write(r.content)\n",
+    "                print(f\"Downloaded {download_file} into {DATA_FOLDER}/raw/.\")\n",
+    "\n",
+    "                with gzip.open(\n",
+    "                    f\"/lakehouse/default/{DATA_FOLDER}/raw/{download_file}\", \"rb\"\n",
+    "                ) as fin:\n",
+    "                    with open(\n",
+    "                        f\"/lakehouse/default/{DATA_FOLDER}/raw/{DATA_FILE}\", \"wb\"\n",
+    "                    ) as fout:\n",
+    "                        fout.write(fin.read())\n",
+    "                print(f\"Unzip {download_file} into {DATA_FOLDER}/raw/{DATA_FILE}.\")\n",
+    "            except Exception as e:\n",
+    "                print(f\"Failed on downloading {DATA_FILE}, error message: {e}\")\n",
+    "        else:\n",
+    "            print(f\"{DATA_FILE} already exists in {DATA_FOLDER}/raw/.\")"
```

Comment:
We can make this a single line of spark readers if we save this loaded dataset to our public blob storage. I let others slip by with this large code, but are you up to show them how its done :) ?